back/app/models/static_page.rb

```diff
@@ -74,6 +74,11 @@ class StaticPage < ApplicationRecord
     validates :banner_cta_button_url, presence: true, url: true
   end
 
+  validates :banner_signed_in_image_enabled, inclusion: [true, false]
+  with_options if: -> { banner_signed_in_image_enabled == false } do
+    validates :banner_layout, presence: true, inclusion: %w[fixed_ratio_layout]
```

Comment:
Maybe I wouldn't be so strict about it and let the FE decide how to manage `banner_signed_in_image_enabled` and `banner_layout`. It doesn't affect any BE logic, and when this requirement changes, it may be easier just to change the FE.

But both options are fine.